src/lib/data/alignment.js

```diff
@@ -374,13 +425,11 @@ export default class Alignment {
    * @param {Token} token
    * @returns {Boolean} true - groups were merged, false - was not
    */
-  mergeActiveGroupWithAnotherByToken (token) {
-    if (this.hasActiveAlignment && this.tokenIsGrouped(token)) {
-      const tokensGroup = this.findAlignmentGroup(token)
+  mergeActiveGroupWithAnotherByToken (token, limitByTargetId) {
+    if (this.hasActiveAlignmentGroup && !this.tokenInActiveGroup(token, limitByTargetId) && this.tokenIsGrouped(token, limitByTargetId)) {

```

Comment:
There is a bug here (i think). The segment is not checked when merging two groups. To reproduce the bug: 

click origin text token A Segment 1
click target text token B Segment 1
click origin text token A Segment 1(alignment group is saved)
click origin text token C Segment 2
click origin text token A Segment 1
click origin text token C Segment 2
The alignment group now contains Origin text Token A from Segment 1 Token C from Segment 2 and Target Text Token B from Segment 1

Reply:
Great, thank you for the catch!

Reply:
Fixed